service_provider.go

```diff
@@ -1251,21 +1256,42 @@ func (r *LogoutRequest) Redirect(relayState string) *url.URL {
 
 	rv, _ := url.Parse(r.Destination)
 
-	query := rv.Query()
-	query.Set("SAMLRequest", w.String())
+	query := rv.RawQuery
```

Comment:
Are you intentionally building URL query manually?

Reply:
It came from the solution from github issue.

There's some base64 -> url encoding that will likely causes problems.